src/dag.c

```diff
@@ -576,7 +574,7 @@ int RAI_parseDAGPersistArgs(RedisModuleCtx *ctx, RedisModuleString **argv,
       separator_flag = 1;
       break;
     } else {
-      AI_dictAdd(*persistContextDict, (void*)arg_string, (void *)1);
+      AI_dictAdd(*persistContextDict, (void*)RedisModule_Strdup(arg_string), (void *)1);
```

Comment:
I've seen that you added Strdup to all keys, but I'm pretty sure they are already dup'ed internally when calling dictAdd

Reply:
You're totally right Luca. Fixed in the latest commit